converter/src/main/java/gov/cms/qpp/conversion/decode/QrdaDecoderEngine.java

```diff
@@ -274,8 +274,9 @@ private boolean containsClinicalDocumentTemplateId(Element rootElement) {
 		List<Element> clinicalDocumentChildren = rootElement.getChildren(TEMPLATE_ID, rootElement.getNamespace());
 
 		for (Element currentChild : clinicalDocumentChildren) {
-			if (getTemplateId(currentChild) == TemplateId.CLINICAL_DOCUMENT) {
-				containsTemplateId = true;
+			TemplateId templateId = getTemplateId(currentChild);
+			if (templateId == TemplateId.CLINICAL_DOCUMENT) {
+				containsTemplateId = templateId.getExtension().equals(currentChild.getAttributeValue(EXTENSION_STRING));
```

Comment:
@halprin: Here's where the extension check is for clinical document.  So, it's still here.

Reply:
Right, the code does work, but I was thinking that the strict checking for a clinical document could remain in `TemplateId` instead of here.